mutation_partition.cc

```diff
@@ -680,7 +680,7 @@ void appending_hash<row>::operator()(Hasher& h, const row& cells, const schema&
     for (auto id : columns) {
         const cell_and_hash* cell_and_hash = cells.find_cell_and_hash(id);
         if (!cell_and_hash) {
-            return;
+            continue;
         }
```

Comment:
Do we need a cluster feature for this? Without a cluster feature, we'll have a read repair storm if many reads have a null in a non-last column.

Reply:
Also, won't it break schema digests? They also use appending_hash.

Reply:
Adding @tgrabiec for an opinion.

Reply:
Aside from cluster features, there's also `schema_features` from `schema_features.hh`, they might be applicable - one of them was already introduced to guard schema digest calculations between incompatible versions.

Reply:
Yes, but you need a cluster feature to select the correct schema featuress.

Reply:
A cluster bug will reduce the intensity of the storm, but not prevent it completely since the nodes may learn that we are ready to not be compatible with the broken behaviour at different points in time.

As for schema digests, I guess that depends whether we can get nulls in the middle of a row in the relevant system tables.

Reply:
We have some map types, don't know if empty maps are NULL or {}.

Reply:
About affect on a schema digest. This fixes is a regression. If there regression did not cause problems with schema sync I do not see how the fix will.

Reply:
The same is true for read repairs. If they were not a problem when the regression was introduced it should not be a problem when it is fixed.

Reply:
I agree with @gleb-cloudius here, simply fixing the regression seems like the best way out of this situation. This pull request looks quite complex due to code movements, but it's effectively `s/return/continue/g` + a unit test.